examples/demo-stf/src/sov-cli/native.rs

```diff
@@ -0,0 +1,487 @@
+            let mut data_list: Vec<Vec<u8>> = vec![];
+
+            for path in path_list {
+                let mut f = File::open(path).expect("Unable to open file");
+                let mut hex_string = String::new();
+                f.read_to_string(&mut hex_string)
+                    .expect("Unable to read the file");
+
+                // decode the hex string to bytes
+                let bytes = hex::decode(hex_string.trim()).expect("Decoding failed");
+
+                // push it into data_list
+                data_list.push(bytes);
+            }
+            let blob = data_list.try_to_vec().unwrap();
+            println!("{}", hex::encode(blob))
+        }
+        Commands::Util(util_args) => match util_args.command {
+            UtilCommands::DeriveTokenAddress {
+                token_name,
+                sender_address,
+                salt,
+            } => {
+                let sender_address = Address::from(
+                    AddressBech32::try_from(sender_address.clone()).unwrap_or_else(|e| {
+                        panic!(
+                            "Failed to derive pub key from string: {}: {}",
+                            sender_address, e
+                        )
+                    }),
+                );
+                let token_address =
+                    sov_bank::create_token_address::<C>(&token_name, sender_address.as_ref(), salt);
+                println!("{}", token_address);
+            }
+
+            UtilCommands::ShowPublicKey { private_key_path } => {
+                let sender_priv_key = SerializedTx::deserialize_priv_key(private_key_path)
+                    .expect("Failed to get private key from file");
+                let sender_address: Address = sender_priv_key.pub_key().to_address();
+                println!("{}", sender_address);
+            }
+
+            UtilCommands::CreatePrivateKey { priv_key_path } => {
+                PrivKeyAndAddress::generate_and_save_to_file(priv_key_path.as_ref())
+                    .unwrap_or_else(|e| panic!("Create private key error: {}", e));
+            }
+            UtilCommands::PrintNamespace => {
+                println!("{}", hex::encode(ROLLUP_NAMESPACE_RAW));
+            }
+        },
+        Commands::Module(module_args) => {
+            let sender_priv_key_path = &module_args.private_key.unwrap();
+            let nonce = module_args.nonce.unwrap();
+
+            let serialized_call: Vec<u8> = module_parse_helper(module_args.command);
+            let serialized =
+                SerializedTx::new_from_call(sender_priv_key_path, serialized_call, nonce)
+                    .unwrap_or_else(|e| panic!("Call message serialization error: {}", e));
+
+            let raw_contents = hex::encode(serialized.raw.data);
+            println!("{}", raw_contents);
+        }
+    }
+}
+
+#[cfg(not(feature = "native"))]
+pub fn main() {
+    println!("build binary with native");
```

Comment:
Why do we have `fn main` here and `main.rs` in the same directory?

Reply:
I did this primarily to ensure sov-cli only runs with the native flag supplied, otherwise it was leading to a lot of conditional annotation on different imports, structs and enums - the code was full #[cfg(feature = "native")] - and since we need native for the entire binary, I just wrapped it in a module and i'm importing that only when native feature is enabled. I tried checking if there are ways in Cargo.toml to restrict that a binary can only be built with specific features but didn't find anything there